internal/profile/install_profile.go

```diff
@@ -0,0 +1,194 @@
+	}
+	if exists {
+		localChanges, err := profile.localFilesChanged()
+		if err != nil {
+			return errors.Wrapf(err, "error checking for changes in %s", profile.ProfilePath)
+		}
+
+		// If there's changes and we've selected CreateNew, move the old path
+		// TODO: do we want this to pe appended with some kind of version string instead?
+		if localChanges && !options.OverwriteExisting {
+			if localChanges && options.Name == DefaultProfile {
+				logger.Warn("WARNING: default profile has been changed by user or updated by elastic-package. The current profile will be moved to default_old.")
+			}
+			err = updateExistingDefaultProfile(options.PackagePath)
+			if err != nil {
+				errors.Wrap(err, "error moving old profile")
+			}
+			err = os.Mkdir(profile.ProfilePath, 0755)
+			if err != nil {
+				return errors.Wrapf(err, "error crating profile directory %s", profile.ProfilePath)
+			}
+		}
+	} else {
+		os.Mkdir(profile.ProfilePath, 0755)
+		if err != nil {
+			return errors.Wrapf(err, "error crating profile directory %s", profile.ProfilePath)
+		}
+	}
+	if err != nil {
+		return errors.Wrapf(err, "stat file failed (path: %s)", profile.ProfilePath)
+	}
+
+	//write the resources
+	return profile.writeProfileResources()
+
+}
+
+// updateExistingDefaultProfile migrates the old default profile to profile_old
+func updateExistingDefaultProfile(path string) error {
+	profile, err := NewConfigProfile(path, DefaultProfile)
+	if err != nil {
+		return errors.Wrap(err, "error creating profile")
+	}
+	meta, err := profile.metadata()
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+	meta.Name = "default_old"
+	err = profile.updateMetadata(meta)
+	if err != nil {
+		return errors.Wrap(err, "error updating metadata")
+	}
+	err = os.Rename(profile.ProfilePath, filepath.Join(profile.ProfilePath+"_old"))
+	if err != nil {
+		return errors.Wrap(err, "error moving default profile")
+	}
+
+	return nil
+}
+
+// createProfileFrom creates a new profile by copying over an existing profile
+func createProfileFrom(option Options) error {
+	fromProfile, err := loadProfile(option.PackagePath, option.FromProfile)
+	if err != nil {
+		return errors.Wrapf(err, "error loading %s profile", option.FromProfile)
+	}
+
+	newProfile, err := NewConfigProfile(option.PackagePath, option.Name)
+	if err != nil {
+		return errors.Wrapf(err, "error creating %s profile", option.Name)
+	}
+
+	newExists, err := newProfile.alreadyExists()
+	if err != nil {
+		return errors.Wrapf(err, "error checking profile %s", newProfile.ProfilePath)
+	}
+	if newExists {
+		return errors.Errorf("profile %s already exists", newProfile.profileName)
+	}
+	os.Mkdir(newProfile.ProfilePath, 0755)
+
+	newProfile.overwrite(fromProfile.configFiles)
+	return newProfile.writeProfileResources()
+
+}
+
+// LoadProfileFromDefaultLocation loads an existing profile from the default elastic-package config dir
+func LoadProfileFromDefaultLocation(profileName string) (*Profile, error) {
```

Comment:
Can we drop the suffix `...FromDefaultLocation`? There is no other option available, so it's a bit confusing.